File: app/src/main/java/com/designpatterns/demo/creational/builder/java/Builder.java
package com.designpatterns.demo.creational.builder.java;

/**
 * 抽象建造者，创建产品实例并返回
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/5/10 下午4:43
 * @project_name DesignPatternsDemo
 */
public abstract class Builder {

    public abstract void buildName();

    public abstract void buildPrice();

    public abstract Product getProduce();


}

File: app/src/main/java/com/designpatterns/demo/builder/kotlin/Builder.kt
package com.designpatterns.demo.builder.kotlin

/**
 *
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/5/14 下午1:29
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */

abstract class Builder{

    abstract fun buildName()

    abstract fun buildPrice()

    abstract fun getProduce():Product
}
